### DISCUSSION TITLE
Camera Orientation

### DISCUSSION THREAD

**Discussion by sebGuerrero** [enebin/Aespa#33] [Q&A] [answered]
Hey guys I think I have the most simple setup

```
import Aespa
import SwiftUI

struct VideoContentView: View {
  @StateObject private var viewModel = VideoContentViewModel()
  @State var pvId = UUID()
  var body: some View {
    VStack{
      viewModel.preview
        .frame(minWidth: 1000,
               maxWidth: .infinity,
                 minHeight: 0,
                 maxHeight: .infinity)
          .edgesIgnoringSafeArea(.all)
        
        Button {
          viewModel.rotate()
        } label: {
          Text("ROTATE")
            .background(.red)
        }

      }
    }
}

class VideoContentViewModel: ObservableObject {
  let aespaSession: AespaSession
  var preview: some View {
    aespaSession.interactivePreview()

  }
  
  init() {
    let option = AespaOption(albumName: "YOUR_ALBUM_NAME")
    self.aespaSession = Aespa.session(with: option)
    aespaSession
      .common(.quality(preset: .high))
  }
  
  func rotate() {
    aespaSession.common(.orientation(orientation: .landscapeRight))
  }
}
```

I want the view and camera to follow the rotation of the device. 
When rotating the phone the camera does not rotate until you press rotate (which is fine for my use case) but the problem is with the preview that does not occupy the full width when on landscape.

What am I missing?

**Comment by enebin** [comment, marked as answer]
Hi, @sebGuerrero 👋. Thank you for your question!

First, note that the `orientation` setting in `common` is not for the preview's direction but for setting the direction of the saved video. There seems to have been some misunderstanding about this part.

The orientation of the preview basically does not change. This is intentional. According to the default settings provided by `AVFoundation`, rotating the orientation would also rotate the preview, resulting in a broken image. Therefore, the preview has been set not to rotate separately.

The reason why the preview does not fully fill the screen in the code you provided appears to be because you have set the `minWidth` to `1000` in the `frame`. When the width becomes `1000` and the preview aligns with the frame's leading edge, it seems that only the far right of the preview is being displayed while the left part is cut off. To resolve this, try changing the `minWidth` to `0`. Then, the preview should appear to fully fill the screen.

If this does not resolve your problem, please comment to this reply. Additional questions are always welcome!

**Reply by sebGuerrero** [reply]
Hey @enebin thanks for the reply. I copied the wrong code, I indeed have 0 as minWidth, and still when rotating the device I can see only half of the view

**Reply by sebGuerrero** [reply]
Even from your demo app if I enable different orientations for the app we'll have the same issues:
<img width="279" alt="image" src="https://github.com/enebin/Aespa/assets/5124440/3bae86f3-ccc0-438f-baa8-549db48f5d15">

In my case I need to support multiple orientations for other components of my need to react to the device's orientation. Do you have any suggestions for this?

**Reply by sebGuerrero** [reply]
I went through the docs and just understood what you mention about the orientation of the preview not changing and this being intentional, It makes sense to me now.

**Comment by subelsky** [comment]
fyi  #43 should help render preview layers more accurately when orientation changes. Though also see the caveat about the need for application code to update the session's orientation.
